R/qualification-ddi.R

```diff
@@ -242,43 +242,38 @@ buildQualificationDDIDataframe <- function(dataframe,
 #' @param dataVector a numeric vector containing the x or y values of datapoints to be plotted
 #' @return a list containing the minimum and maximum axes limits
 #' @keywords internal
-getSmartZoomLimits <- function(isResidualsVsObserved,dataVector){
-
+getSmartZoomLimits <- function(isResidualsVsObserved, dataVector){
   validateIsLogical(isResidualsVsObserved)
   validateIsNumeric(dataVector)
 
+  # Guest criterion range for obs vs pred DDI Ratio plot
   if(!isResidualsVsObserved){
-
-    if(min(dataVector) >= 1){
-      minLimit <- min(0.8*min(dataVector), 0.8)
-    } else {
-      minLimit <- min(0.8*min(dataVector), 0.25)
-    }
-
-    if(min(dataVector) >= 1){
-      maxLimit <- max(1.25*max(dataVector), 4)
-    } else {
-      maxLimit <- max(1.25*max(dataVector), 1.25)
-    }
-
+    # Use ifelse to simplify notations and clarify what code does
+    minLimit <- ifelse(
+      min(dataVector)>=1,
+      min(0.8*min(dataVector), 0.8),
+      min(0.8*min(dataVector), 0.25)
+    )
+    maxLimit <- ifelse(
+      max(dataVector)>=1,
+      max(1.25*max(dataVector), 4),
+      max(1.25*max(dataVector), 1.25)
+    )
```

Comment:
Why do we need the lines 249-261?
(minLimit, maxLimit) returned here is the same as in the lines 265-276 below.

Reply:
Actually we don't !!
The code was mirroring the Matlab code, which was formulated this way because the function was using user-defined limits instead of min/max of data.